src/classes/TelemetryQueueable.cls

```diff
@@ -0,0 +1,81 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+public without sharing class TelemetryQueueable implements System.Queueable {
+    @TestVisible
+    private List<Telemetry> telemetries;
+
+    public void enqueue(List<Telemetry> telemetries) {
```

Comment:
I like your queueable approach. Keep in mind that each queueable transaction will be limited to a max of 50K rows in total (though should be fine with `Count()`) and a cpu time of 60 seconds. Depending on the size of the objects being counted it's possible those queries could time out, especially if a single queueable is running multiple count queries. I ended up moving counts against Opps and Recurring Donations into a batch job, which now I have to tweak to chunk in batches of 10M to avoid an error in the one customer that has 80M opps.

I see you're using the -1 default value for the feature parameter integers, so if there are any timeouts/failures you should be able to identify the orgs (or at least the number of orgs) by running a report of them in the LMO. Worse case, you can adjust those queries in an upcoming iteration.

Reply:
We are only doing counts for all of our numbers, if the count exceeds the 50k we should be ok right? I know we tested count with million+ in dso so assumed it'd be ok.

Reply:
The 50K limit shouldn't be an issue at all, but if you're doing more than one count in a single operation the timeout is a possibility. Generally I agree that counting 1M records should be ok, though I've occasionally seen that fail (rare as it may be).

Reply:
We have Contacts in its own job but we do have 2 separate counts so we will definitely watch for the -1's and make adjustments in future runs if we notice failures. Thanks!